connection_maker.go

```diff
@@ -100,7 +101,9 @@ func (cm *ConnectionMaker) InitiateConnections(peers []string, replace bool) []e
 			host = peer
 			port = "0" // we use that as an indication that "no port was supplied"
 		}
-		if addr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("%s:%s", host, port)); err != nil {
+		if !isAlnum(port) {
+			errors = append(errors, fmt.Errorf("invalid peer name '%s', should just be host[:port]", peer))
+		} else if addr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("%s:%s", host, port)); err != nil {
```

Comment:
I know.  (a) the `Sprintf` was there before; cleaning up arbitrary deficiencies is not the intention of this change
(b) I didn't realise `ResolveTCPAddr` needs a port number, so ignore the bit about `peer`.